### ISSUE TITLE
When changing filter values, the sort direction changes every time you click 

### ISSUE THREAD

**Issue by jeznag** [react-bootstrap-table/react-bootstrap-table2#380]
When I change the filter values, the sort direction changes at the same time even though I didn't click on the sort caret.
![filtering sorting](https://user-images.githubusercontent.com/1034890/41519103-6e9a6fb2-7309-11e8-8565-5045aaea22d7.gif)

https://jeznag.github.io/react-bootstrap-table2-example/index.html

https://github.com/jeznag/react-bootstrap-table2-example/blob/master/src/DataJunction.js

Code is basically:
```
class DataJunction extends Component {
  constructor(props) {
    super(props);
    this.state = {
      rows: manyRows.slice(0)
    };
  }

  render() {
    return (
      <BootstrapTable
        keyField="id"
        data={this.state.rows}
        columns={columns}
        filter={filterFactory()}
        pagination={paginationFactory()}
        selectRow={{
          mode: "checkbox",
          clickToSelect: true
        }}
      />
    );
  }
}
```

**Comment by AllenFang** [comment]
@jeznag Thanks for this bugs report, I will check this out and fix it ASAP 👍

**Comment by AllenFang** [comment]
@jeznag 

[RELEASE NOTES](https://react-bootstrap-table.github.io/react-bootstrap-table2/blog/2018/06/24/version-bump.html)

Fixed on `react-bootstrap-table-next@0.1.14` and `react-bootstrap-table2-filter@0.3.0`, please re-open this issue and tag me if the problem still remain
